common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -34,6 +34,11 @@
   public final int days;
   public final long microseconds;
 
+  // Previous CalendarInterval has 2 fields: months and microseconds. Here we add one more days
```

Comment:
Previous -> Previously,
has -> had
Nit: I don't know if it matters to talk about the previous implementation. A note about why days exist is fine.
Frankly it might also be worth a note about why months exists at all (they are also not units of time with a constant length, unlike hours, minutes, seconds)

Reply:
@srowen thanks for your suggestion, I changed another way to note about the reason. Could you review again?